### ISSUE TITLE
arcs dies with error: `@HD     VN:1.3  SO:coordinate': No such file or directory

### ISSUE THREAD

**Issue by claumer** [bcgsc/arcs#89]
Hi there,

I'm trying to scaffold a PacBio genome assembly made with wtdbg2 using some 10X reads. I've mapped the reads using longranger align, and am pointing to their bam file as such:

arcs -a /nfs/research1/marioni/claumer/longranger-2.2.2/DalyG_longranger/outs/possorted_bam.bam -b DalyG_redbean_arcs

Have also tried with arks mode:

arcs --arks -t 40 -f ../DalyG_HiFi/DalyG_HiFi_q10_redbean_all.ctg.fa -a ../longranger-2.2.2/DalyG_longranger/outs/possorted_bam.bam

And basically, the job runs a while, uses a lot of memory, and then dies, pumping to STDERR the subject line of this github issue. Can you give me some idea of what the problem might be? Could it be that longranger provides a coordinate-sorted bam file, and arcs needs an unsorted one?

Regards,
Chris L

**Comment by lcoombe** [comment]
Hi @claumer,

How is the `longranger align` BAM file sorted? ARCS expects the input alignment file to be sorted by name, and have the barcode in the BX tag. Take a look at our `arcs-make` Makefile, which takes the draft assembly and the post-longranger basic reads as input. That Makefile runs the relevant alignments, arcs, LINKS steps so you can see the expected format of the files at each step.

As for the `--arks` mode, this mode uses exact kmer matching instead of alignments to map barcodes to contig ends, so if you want to use ARCS in this mode, you would need to input the reads, not the alignment file. Again, I'd recommend taking a look at the `arcs-make` Makefile which is designed to make running the pipeline simple.

Hope that helps!
Lauren

**Comment by claumer** [comment]
Hi Lauren,

Thanks for your quick response. The original longranger bam file I think was coordinate sorted, but I just used samtools to resort it with -n (sorting by name), and ran arcs again on this sorted file, and ran into exactly the same form of error. The reads do have the BX tag in their read names, I can confirm. Any other ideas of what could be going wrong here?

I would think that barcode-aware mapping should be the ideal strategy, so that's why I was using longranger... but I will indeed have a gander at the makefile, and see if I can make any sense of it and see where those formats differ from my data, especially if we can't solve why the longranger bam is failing here.

Regards,
Chris L

**Comment by lcoombe** [comment]
Based on your above command, it looks like you are directly inputting the BAM file with `-a` - If you specify the alignments with this option, it expects a text file listing each input BAM file on separate lines. If you want to just use the single BAM file, you can list the file name as a positional argument instead.

**Comment by claumer** [comment]
Hi Lauren,

OK, I think I see the problem. I got confused by the documentation:

(ARCS) [claumer@noah-login-03 DalyG_GyrF_10X]$ arcs --help | head -n 30
Reading user inputs...
arcs 1.1.1

Usage: arcs [Options] <list of alignment files>  or
       arcs [Options] --arks -f <contig sequence file> <list of linked read files>


which implies that I can point ARCS to a list of alignment files positionally at the top (as above), but then later says I need to use the -a flag to do so. When I use "arcs fileoffilenames" I get:

(ARCS) [claumer@noah-login-03 DalyG_GyrF_10X]$ arcs DalyG_fofn.txt
Reading user inputs...
Unknown type file is observed!
Input files must be all alignment or all read files.. Exiting...
File type must be compatible with the method. (BAM/SAM for ARCS) or (Read file for ARKS (--arks)). Exiting...
Try arcs --help for more information.

Indeed, it seems that using "arcs -a fileoffilenames" or "arcs bamfile" causes the program to run maybe as expected (stdout below):

--

Reading user inputs...
Finished reading user inputs...entering runArcs()...
Running: arcs 1.1.1
ARCS method
 pid 283715
 -c 5
 -d 0
 -e 30000
 -l 0
 -m 50-10000
 -r 0.05
 -v 0
 -z 500
 --gap=100

--

What would help prevent such confusion in the future would be to a.) fix the documentation and b.) add more useful error messages and/or format-checking preflight. ARCS did this successfully when I ran arcs fofn.txt without an -a flag, but it failed to pick up on it when I did the inverse, gave it a path to a bam file through the -a flag.

Keen to see how this works with my data, though! Thanks for your input on figuring this out.

Regards,
Chris L
